File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
    private static readonly string[] _colors = new[] { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };
    public static int ColorCode(string color) => Array.IndexOf(_colors, color);

    public static string[] Colors() => _colors;
}

File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.List;

class ResistorColor {
    private static final List<String> _colors = List.of("black", "brown", "red", "orange", "yellow", "green", "blue",
            "violet", "grey", "white");

    int colorCode(String color) {
        return _colors.indexOf(color);
    }

    String[] colors() {
        return _colors.toArray(new String[0]);
    }
}
